.changeset/purple-olives-grin.md

```diff
@@ -0,0 +1,6 @@
+---
+'@shopify/polaris': patch
+'polaris.shopify.com': patch
+---
+
+Added `disabled` property to the `Row` component to determine whether the row should have border-top
```

Comment:
Nit, and it would be best to consult the Polaris team on this.

Having a `divider` prop makes me think we can configure the divider by passing a certain value to it, e.g. a token or [divider](https://polaris.shopify.com/components/layout-and-structure/divider) component which is not the case and might be misleading. 

In our case, it's a boolean that toggles the visibility of the divider, looking at other polaris components, it's usually used like `labelHidden`, so we might want to change it to `dividerHidden` or something similar.

Or we could change it to `borderless` which would be also descriptive.